### PR TITLE
Add a demo map viewer for QtWidgets

### PR DESCRIPTION
**Pull request by syyyr**
In our app, we only need a simple map viewer that doesn't use QML. It seems to me that a QtWidgets demo app was missing from _libosmscout_, so I made one.

This map viewer is very simple. It shows it in a window and supports moving the map with the mouse cursor and zooming with the mouse wheel.

### CODE REVIEW

**Comment by syyyr** [comment]
It seems that CI compiles the project with Qt 5.12. Is this version still supported by libosmscout? Should I port my PR to that version?

**Comment by Framstag** [comment]
@Karry What is your opinion?

**Review by Karry — CHANGES_REQUESTED** [review]
Hi. I like it in general :-) 

In client-qt library, I am keeping compatibility with Qt 5.6, because of slow progress in Sailfish OS. But we don't have to keep it in all demos. In other hand, test the build on Ubuntu 20.04 is beneficial IMHO. So, you may add check to CMake and Meson for Qt version and skip this demo with old Qt, or just add `#if QT_VERSION < 0x051400` and use `QWheelEvent::pos()` that is deprecated now... 

I do not like code that is full of macros, but there is no other option when we support range of Qt versions with evolving api.

File: Demos/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +46,7 @@ osmscout_demo_project(NAME Routing SOURCES src/Routing.cpp TARGET OSMScout::OSMS
 #---- RoutingAnimation
 if(${OSMSCOUT_BUILD_MAP_QT})
 	osmscout_demo_project(NAME RoutingAnimation SOURCES src/RoutingAnimation.cpp TARGET OSMScout::OSMScout OSMScout::Map OSMScout::MapQt Qt5::Widgets)
+	osmscout_demo_project(NAME QtWidgetsDemoApp SOURCES src/QtWidgetsDemoApp.cpp TARGET OSMScout::ClientQt)
```
[the remark this answers is not in the document]
Comment (syyyr) [reply Demos/CMakeLists.txt]: Done

**Comment by syyyr** [comment]
I moved the CMake declaration and added an ifdef for pos(). Hopefully it'll build now.

**Review by Framstag — APPROVED** [review]
Looks OK for me.

**Review by Karry — APPROVED** [review]

**Pull request merged**